lib/collection/src/shard/shard_holder.rs

```diff
@@ -64,6 +71,55 @@ impl ShardHolder {
         shard_ops
     }
 
+    /// Create directory to hold the temporary data for shard with `shard_id`
+    pub async fn create_temporary_shard_dir(
+        collection_path: &Path,
+        shard_id: ShardId,
+    ) -> CollectionResult<PathBuf> {
+        let shard_path = collection_path.join(format!("{shard_id}-temp"));
+        tokio::fs::create_dir_all(&shard_path)
+            .await
+            .map_err(|err| CollectionError::ServiceError {
+                error: format!(
+                    "Can't create shard {shard_id} temporary directory. Error: {}",
+                    err
+                ),
+            })?;
+        Ok(shard_path)
+    }
+
+    /// Initiate temporary shard
+    pub async fn initiate_temporary_shard(
+        &mut self,
+        collection_id: CollectionId,
+        collection_path: &Path,
+        shared_config: Arc<RwLock<CollectionConfig>>,
+        shard_id: ShardId,
+    ) -> CollectionResult<()> {
+        // check if a temporary shard is already present
+        if self.temporary_shards.contains_key(&shard_id) {
```

Comment:
Let's stick to the idempotent property - if operation performed multiple times, the result should be same as if it was performed only once. Besides, I would expect that it case of multiple attempts to transfer shard we would need to initialize it multiple times.

Reply:
So we would accept any subsequent initiate requests without touching the content of the temporary shard?
What was transferred stays in place.

Reply:
I kept a log statement for observability https://github.com/qdrant/qdrant/pull/829/commits/1526c3d8d0f1014e9e9c7fafd65b4eebe6c17fc8